examples/event_handling/lasso_demo.py

```diff
@@ -84,6 +86,11 @@ def on_press(self, event):
         # acquire a lock on the widget drawing
         self.canvas.widgetlock(self.lasso)
 
+    def on_release(self, event):
+        if self.canvas.widgetlock.locked():
+            self.canvas.widgetlock.release(self.lasso)
+            return
```

Comment:
You need to also check that the lasso is the current owner otherwise you risk throwing an error https://github.com/matplotlib/matplotlib/blob/39070aeaab732eca1b5b7f7b478fa8b72dab8507/lib/matplotlib/widgets.py#L50:

```suggestion
        if self.canvas.widgetlock.isowner(self.lasso):
            self.canvas.widgetlock.release(self.lasso)
```